src/sentry/static/sentry/app/views/performance/vitalDetail/utils.tsx

```diff
@@ -0,0 +1,116 @@
+import {Query, Location} from 'history';
+
+import {getAggregateAlias, WebVital} from 'app/utils/discover/fields';
+import {decodeScalar} from 'app/utils/queryString';
+import {Series} from 'app/types/echarts';
+
+import {WEB_VITAL_DETAILS} from '../transactionVitals/constants';
+
+export function generateVitalDetailRoute({orgSlug}: {orgSlug: string}): string {
+  return `/organizations/${orgSlug}/performance/vitaldetail/`;
+}
+
+export const vitalsThresholdFields = {
+  [WebVital.FP]: 'count_at_least(measurements.fp, 3000)',
+  [WebVital.FCP]: 'count_at_least(measurements.fcp, 3000)',
+  [WebVital.LCP]: 'count_at_least(measurements.lcp, 4000)',
+  [WebVital.FID]: 'count_at_least(measurements.fid, 300)',
+  [WebVital.CLS]: 'count_at_least(measurements.cls, 0.25)',
+};
+export const vitalsBaseFields = {
+  [WebVital.FP]: 'count_at_least(measurements.fp, 0)',
+  [WebVital.FCP]: 'count_at_least(measurements.fcp, 0)',
+  [WebVital.LCP]: 'count_at_least(measurements.lcp, 0)',
+  [WebVital.FID]: 'count_at_least(measurements.fid, 0)',
+  [WebVital.CLS]: 'count_at_least(measurements.cls, 0)',
+};
+
+export function vitalDetailRouteWithQuery({
+  orgSlug,
+  vitalName,
+  projectID,
+  query,
+}: {
+  orgSlug: string;
+  vitalName: string;
+  query: Query;
+  projectID?: string | string[];
+}) {
+  const pathname = generateVitalDetailRoute({
+    orgSlug,
+  });
+
+  return {
+    pathname,
+    query: {
+      vitalName,
+      project: projectID,
+      environment: query.environment,
+      statsPeriod: query.statsPeriod,
+      start: query.start,
+      end: query.end,
+      query: query.query,
+    },
+  };
+}
+
+export function vitalNameFromLocation(location: Location): WebVital {
+  const _vitalName = decodeScalar(location.query.vitalName);
+
+  const vitalName = Object.values(WebVital).find(v => v === _vitalName);
+
+  if (vitalName) {
+    return vitalName;
+  } else {
+    return WebVital.FID;
+  }
+}
+
+export function getVitalDetailTableStatusFunction(vitalName: WebVital): string {
+  const vitalThreshold = WEB_VITAL_DETAILS[vitalName].failureThreshold;
+  const statusFunction = `compare_numeric_aggregate(${getAggregateAlias(
+    `p75(${vitalName})`
+  )},greater,${vitalThreshold})`;
+  return statusFunction;
+}
+
+export const vitalMap: Partial<Record<WebVital, string>> = {
```

Comment:
All these names are already defined in this [file](https://github.com/getsentry/sentry/blob/b09bd2d62edf37a9723eff277fc49bcc8c41e3a0/src/sentry/static/sentry/app/views/performance/transactionVitals/constants.tsx#L12), can it be imported instead of duplicated here?

Reply:
ditto the other comment